modules/dice/roll-builder.js

```diff
@@ -135,7 +135,7 @@ export default class RollBuilderFFG extends FormApplication {
       }
 
       // validate that required data is present
-      if (this.roll.hasOwnProperty('item') && !this.roll.item.flags.starwarsffg.uuid) {
+      if (this.roll.hasOwnProperty('item') && this.roll.item.hasOwnProperty('flags') && !this.roll.item.flags.starwarsffg.uuid) {
```

Comment:
I think you could potentially also handle this with nullish checks on the uuid...

```suggestion
      if (this.roll.hasOwnProperty('item') && !this.roll.item?.flags?.starwarsffg?.uuid) {
```

Reply:
I tried that - because it accesses the UUID after checking if it's got a value, the nullish check actually causes issues without fixing anything

Reply:
Ah strange, it should just essentially function the same as the hasOwnProperty check, as it translates to:
```
if (this.roll.hasOwnProperty('item') && this.roll.item.hasOwnProperty('flags') && this.roll.item.flags.hasOwnProperty('starwarsffg') && !this.roll.item.flags.starwarsffg.uuid)
```

Just safely accesses each property before proceeding, ending prematurely at any stage.

The result just means that you have to treat any of the elements as potentially null or undefined in the rest of the business logic.